### PR TITLE
Leaf 3221/3210 stop removal of locally managed and dealing with unexpected data

### PR DESCRIPTION
**Pull request by jampaul3**

### CODE REVIEW

**Review by pete-nerantzinis — APPROVED** [review]

**Review by raphaelfretter — APPROVED** [review]

File: LEAF_Nexus/sources/Group.php [hunk 1]
```diff
@@ -1082,6 +1082,25 @@ public function removePermission($groupID, $categoryID, $UID, $permissionType)
...
+    /**
+     *
+     * @param string $tag_name
+     * @param int $groupID
+     *
+     * @return void
+     *
+     * Created at: 11/1/2022, 11:29:52 AM (America/New_York)
+     */
+    public function addGroupTag(string $tag_name, int $groupID): void
```
Comment (Pelentan) [inline LEAF_Nexus/sources/Group.php:1094]: Probably want to make sure neither of these are empty.

File: LEAF_Nexus/sources/Group.php [hunk 2]
```diff
@@ -1122,11 +1141,11 @@ private function getUIDDisplay($categoryID, $UID){
 
     private function getPositionDisplay($UID){
```
Comment (Pelentan) [inline LEAF_Nexus/sources/Group.php:1142]: Can you update these functions to bring them up to our standards?

File: LEAF_Request_Portal/sources/System.php [hunk 3]
```diff
@@ -1035,17 +1035,27 @@ public function syncSystem(\Group $org_group, \Service $org_service, \OrgChart\G
...
+            $positions = $nexus_group->listGroupPositions($group['groupID']);
+            $employees = $nexus_group->listGroupEmployees($group['groupID']);
+
+            foreach ($positions as $position) {
+                $employees = array_merge($employees, $nexus_position->getEmployees($position['positionID']));
+            }
 
             foreach ($employees as $employee) {
-                if ($employee['userName'] != '') {
+                if (isset($employee['userName']) && !empty($employee['userName'])) {
```
Comment (Pelentan) [inline LEAF_Request_Portal/sources/System.php:1058]: Why not use the empty() function here?

File: LEAF_Request_Portal/sources/System.php [hunk 4]
```diff
@@ -1055,7 +1065,7 @@ public function syncSystem(\Group $org_group, \Service $org_service, \OrgChart\G
                     $backups = $nexus_employee->getBackups($employee['empUID']);
 
                     foreach ($backups as $backup) {
-                        if ($backup['userName'] != '') {
+                        if (isset($backup['userName']) && !empty($backup['userName'])) {
```
Comment (Pelentan) [inline LEAF_Request_Portal/sources/System.php:1068]: Same here

File: LEAF_Request_Portal/sources/System.php [hunk 5]
```diff
@@ -1109,7 +1118,9 @@ private function processServices(array $portal_services, array $portal_chiefs, a
             } else {
                 // service does not exist add it to the portal db
                 //echo 'The service \'' . $service['service'] . '\' was added.<br/>';
-                $org_service->importService($service['serviceID'], $service['service'], $service['abbreviatedService'], $service['groupID']);
+                if(is_numeric($service['serviceID']) && !empty($service['service']) && (is_numeric($service['groupID']) || is_null($service['groupID']))) {
```
Comment (Pelentan) [inline LEAF_Request_Portal/sources/System.php:1121]: You don't need the empty() function here.  is_numeric() covers that.  A null or empty string returns false.

**Review by Pelentan — CHANGES_REQUESTED** [review]

File: LEAF_Nexus/sources/Group.php (hunk 1 above)
Comment (jampaul3) [reply LEAF_Nexus/sources/Group.php:1094]: The explicit calling of string and int will not allow it to come in empty. What I did do was made the check where the method is called, in System.php

**Review by shaneodd — APPROVED** [review]

**Review by Pelentan — APPROVED** [review]

**Review by aerinkayne — APPROVED** [review]

**Pull request merged**